controllers/database_controller.go

```diff
@@ -549,6 +544,44 @@ func (r *DatabaseReconciler) createProxy(ctx context.Context, dbcr *kciv1alpha1.
 	return nil
 }
 
+func (r *DatabaseReconciler) createConnectionString(ctx context.Context, dbcr *kciv1alpha1.Database) error {
+	// First of all the password should be taken from secret because it's not stored anywhere else
+	databaseSecret, err := r.getDatabaseSecret(ctx, dbcr)
+	if err != nil {
+		return err
+	}
+	// Then parse the secret to get the password
+	databaseCred, err := parseDatabaseSecretData(dbcr, databaseSecret.Data)
+	if err != nil {
+		return err
+	}
+	// Fill ConnectionStringFields struct
+	dbUrl := ConnectionStringFields{
+		DatabaseHost: dbcr.Status.ProxyStatus.ServiceName,
+		DatabasePort: dbcr.Status.ProxyStatus.SQLPort,
+		UserName:     dbcr.Status.UserName,
+		Password:     databaseCred.Password,
+		DatabaseName: dbcr.Status.DatabaseName,
+	}
+	// Generate the connection string
+	dbConnectionString, err := generateConnectionString(dbcr, dbUrl)
+	if err != nil {
+		return err
+	}
+	// Update database-credentials secret.
+	if connectionString := databaseCred.ConnectionString; connectionString != "" && connectionString == dbConnectionString {
```

Comment:
why do you check here `connectionString != ""` ? when `connectionString` is "", if the new string is also "" , you can still skip update.

Reply:
It was for debugging purposes, and I forgot to remove it. I was checking when conn_string is being created or updated. I'll remove the first condition to make the connection string be updated if the new connection string is not equal to the old one. 

```
	if connectionString := databaseCred.ConnectionString; connectionString == dbConnectionString {
		return nil
	}

```

Reply:
Done